todo_app/data/user.py

```diff
@@ -0,0 +1,35 @@
+from flask_login import UserMixin, current_user
+from functools import wraps
+
+ROLES = {
+    'reader': 'reader',
+    'writer': 'writer'    
+}
+
+class User(UserMixin):
+    def __init__(self, user_id, username,email, role=ROLES['reader']):
+        self.id = user_id
+        self.username = username
+        self.email = email
+        self.role = role     
+        
+    def is_writer(self):        
+        return self.role == ROLES['writer']
+    
+    def is_reader(self):
+        return self.role == ROLES['reader']
+
+
+def requires_roles(*roles):    
+    def wrapper(f):
+        @wraps(f)
+        def wrapped(*args, **kwargs):
+            if type(current_user).__name__!='LocalProxy':
```

Comment:
What is this line trying to do? Its preventing your role verification from actually having any effect - my user matches this type, and therefore bypasses your security; I was able to [add a todo to your production app](http://mongoappex10.herokuapp.com/) without access to your Github account

Reply:
Could you check if this has been resolved

Reply:
Doesn't look like it - I can't see anywhere in your app that sets the app config LOGIN_DISABLED value, so it's working out here as `None` rather than False; we could set that on the app when it's created (e.g. from an environment variable, with a default of false), or we could check here for any value that isn't True, i.e. the below should get this working:
```python
if current_app.config.get('LOGIN_DISABLED') != True:
```

One easy way to check if this is working is to access your site as a readonly user and then access the add-todo form (locally you can switch off the `{% if isWriter==True: %}` check in the template, or on your heroku site you can inject the form into the HTML) and then see if you can submit a todo that way

Reply:
I have updated the code and had to restart heroku as it seems it has cached the session variables which I thought should of expired after 20 mins.